juju/juju
# fix: temp fix for serialization issue

There is an issue when attempting to serialize the agent config to a machine that isn't the bootstrapped machine. It manifests its self as:

      line 49: cannot unmarshal !!str `1Âµs` into time.Duration

More work to diagnose this is required, but it's clear that something has changed recently, but I'm not sure what. For now, to unblock this, we can set the controller config value to millisecond, which doesn't break the encoding.

I've also added better error message when we can't read the config as the first one is masked.




## QA steps

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
$ juju status
$ juju ssh 0
$ cat /var/lib/juju/agents/machine-0/agent.conf
```

Ensure that the machine correctly comes up and not just starts, also ensure that the agent.conf is correct for the `opentelemetrytailsamplingthreshold` value. Make sure it's not `opentelemetrytailsamplingthreshold: 1Âµs`
